version/derive/src/lib.rs

```diff
@@ -374,7 +364,7 @@ fn impl_declare_versioned(
         pub enum #enum_name {
             #(
                 /// This variant represents a particular version.
-                #scale_variant_attributes #json_variant_attributes
```

Comment:
@0x009922 was this a bug or a feature?
Look [here](https://github.com/paritytech/parity-scale-codec) for what `codec` does.
E.g.:
```rs
enum VersionedEnum {
    #[codec(index = 1)]
    V1(u32),
    #[codec(index = 2)]
    V2(u64)
```

by default when serialized discriminants would be 0, 1, but with `codec` it is 1, 2

Reply:
Iroha tests are failing now because of this. I can restore it, but tell me what you think

Reply:
I reverted it. Don't want to deal with this now. It only affects how enums are serialized with SCALE

Reply:
I cannot find the reference, but I remember that `Versioned` enums were made to have custom discriminants intentionally. 1 for v1, 2 for v2 etc. So, this is a feature?

Reply:
> I cannot find the reference, but I remember that Versioned enums were made to have custom discriminants intentionally. 1 for v1, 2 for v2 etc. So, this is a feature?

exactly. I've reverted now

Reply:
Actually, I've remembered that the intention was to make the first byte of transferred data reflect the version of it.

Reply:
@mversic, please consider my question before marking as resolved )